lib/client/db/dbcmd/dbcmd.go

```diff
@@ -413,6 +412,33 @@ func (c *CLICommandBuilder) getMongoCommand() *exec.Cmd {
 	return c.exe.Command(mongoBin, args...)
 }
 
+func (c *CLICommandBuilder) getMongoAddress() string {
+	query := make(url.Values)
+
+	// Use 5 seconds for server selection timeout to be more align with the
+	// backend. The environment variable serves as a hidden option to force a
+	// different timeout for debugging purpose or extreme situations.
+	serverSelectionTimeoutMS := "5000"
+	if envValue := os.Getenv(envVarMongoServerSelectionTimeoutMS); envValue != "" {
+		c.options.log.Infof("Using environment variable %s=%s.", envVarMongoServerSelectionTimeoutMS, envValue)
+		serverSelectionTimeoutMS = envValue
+	}
+	query.Set("serverSelectionTimeoutMS", serverSelectionTimeoutMS)
+
+	address := url.URL{
+		Scheme:   "mongodb",
+		Host:     fmt.Sprintf("%s:%d", c.host, c.port),
+		RawQuery: query.Encode(),
+		Path:     fmt.Sprintf("/%s", c.db.Database),
+	}
+
+	// Quote the address for printing as the address contains "?".
+	if c.options.printFormat {
+		return fmt.Sprintf(`"%s"`, address.String())
+	}
+	return address.String()
```

Comment:
Don't we need to quote the connectionString to mitigate shell parsing errors ? 
So instead of invoking `mongosh mongodb://localhost:12345/mydb?serverSelectionTimeoutMS=5000` we   should alywas call `mongosh "mongodb://localhost:12345/mydb?serverSelectionTimeoutMS=5000"` even if TLS routing is disabled ?

Reply:
quoted URL will fail `exec.Cmd.Run` as clients sees the quotation as invalid URL. 

I tested this with Teleport Connect today and I've added `WithPrintFormat` for it to make sure postgres and mongo connection urls are quoted.

after fix:
<img width="665" alt="Screen Shot 2022-06-23 at 10 29 58 AM" src="https://user-images.githubusercontent.com/12417665/175327562-9e6cf0e8-8820-4d7d-ad59-22818fd144b1.png">

I've also added @ravicious for review on that part.